clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -160,92 +228,91 @@ private boolean maybeTransitionToStable() {
         return state.equals(MemberState.STABLE);
     }
 
+    /**
+     * Take new target assignment received from the server and set it as targetAssignment to be
+     * processed. Following the consumer group protocol, the server won't send a new target
+     * member while a previous one hasn't been acknowledged by the member, so this will fail
+     * if a target assignment already exists.
+     *
+     * @throws IllegalStateException If a target assignment already exists.
+     */
     private void setTargetAssignment(ConsumerGroupHeartbeatResponseData.Assignment newTargetAssignment) {
         if (!targetAssignment.isPresent()) {
+            log.debug("Member {} accepted new target assignment {} to reconcile", memberId, newTargetAssignment);
```

Comment:
I am tempted by logging this a info. What do you think?

Reply:
Yes, sounds sensible to me.